README.md

```diff
@@ -47,72 +49,58 @@ You can also initilze a client with a sub-merchant-id:
 
 ```ruby
 client = Bambora::Client.new(
-  api_key: ENV.fetch('BAMBORA_API_KEY'),
-  merchant_id: ENV.fetch('BAMBORA_MERCHANT_ID'),
+  ...
   sub_merchant_id: 'submerchantid',
 )
 ```
 
+The client is then passed to one of the `*Resource` classes, mentioned below, in order to make a network request.
+
 ### Profiles
 
 *Summary*: Payment profiles store confidential payment information. Transactions can be processed against profiles.
 *Bambora Docs*: <https://dev.na.bambora.com/docs/guides/payment_profiles/>
 
+To use the profiles API, you must create an instance of the `Bambora::V1::ProfileResource` class.
+
+```ruby
+client = Bambora::JSONClient.new(...)
+profiles = Bambora::V1::ProfileResource.new(client: client, sub_path: '/v1/profiles')
```

Comment:
This flow applies to all actions for profiles, therefore, brought this up to the top.

Reply:
i was saying you could have a higher level (JSON-)XML*Resource e.g.
```ruby
module Bambora
  module V1
    class JSONProfileResource < ProfileResource
      def initialize(sub_path:)
        super(client: Bambora::JSONClient.new(...), sub_path: sub_path)
      end
    end
  end
end
```
At this point you would build your user public api of "separate bricks"

Reply:
Less boilerplate but if current works for you - im ok with it as well.

Reply:
Yes I agree with this. I contemplated doing this, and decided to just leave it as is, and build these higher-level classes later, in order to narrow the scope of this PR.

Reply:
Should the `*Resource` already know the `sub_path` for it's intended resource? Following the [Principle of Least Astonishment](https://en.wikipedia.org/wiki/Principle_of_least_astonishment), I'd want to pass along as little as possible and have this library handle as much as possible for me.